protocol/contracts/leaser/src/contract.rs

```diff
@@ -199,7 +197,7 @@ fn validate_customer(
 fn validate_lease(lease: Addr, deps: Deps<'_>) -> ContractResult<Addr> {
     Leaser::new(deps)
         .config()
-        .map(|config| config.config.lease_code_id)
+        .map(|config| config.config.lease_code)
         .and_then(|lease_code_id| {
```

Comment:
This can be combined into one `Result::and_then` call to avoid complications. `lease_code_id` is used only once either way, so aliasing the value doesn't matter in this case.

Reply:
indeed it can be but prefer to keep it separate. it's similar to the question how big a function should be